### PR TITLE
feature/31203 - retry logic for all sql consumers

### PR DESCRIPTION
**Pull request by GuyHarwood**
# Add SQL retry logic

- [x] implement for functions 
- [x] implement for admin app directly in sql.service

see [less-tedious v2](https://github.com/DFEAGILEDEVOPS/less-tedious/releases/tag/v2.0.0) for details on how retry was implemented

# New Admin and Function App Environment Variables

`RETRY_MAX_ATTEMPTS` Int defaults to 3
`RETRY_PAUSE_MS` Int defaults to 5000
`RETRY_PAUSE_MULTIPLIER` Float defaults to 1.5

### CODE REVIEW

File: admin/services/data-access/sql.service.js [hunk 1]
```diff
@@ -6,6 +6,20 @@ const dateService = require('../date.service')
 const poolConfig = require('../../config/sql.config')
 const moment = require('moment')
 const logger = require('../log.service').getLogger()
+const retry = require('./retry-async')
+
+const retryConfig = {
+  attempts: 3,
```
Comment (GuyHarwood) [inline admin/services/data-access/sql.service.js]: we should make these overridable via env vars so we can tweak during load tests.

File: functions/check-expiry/v1.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 const sqlService = require('less-tedious')
-const { TYPES } = require('tedious')
+const { TYPES } = sqlService
```
Comment (GuyHarwood) [inline functions/check-expiry/v1.js]: good catch, i forgot to do these 🏆

File: admin/services/data-access/sql.service.js (hunk 1 above)
Comment (jon-shipley) [reply admin/services/data-access/sql.service.js]: Done

Comment (jon-shipley) [reply admin/services/data-access/sql.service.js]: Done

File: functions/disable-functions.env [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+AzureWebJobs.completed-checks.Disabled=true
+AzureWebJobs.expire-prepared-check.Disabled=true
+AzureWebJobs.feedback.Disabled=true
+AzureWebJobs.prepare-check.Disabled=true
+AzureWebJobs.prepared-check-delete.Disabled=true
+AzureWebJobs.prepared-check-sync.Disabled=true
+AzureWebJobs.pupil-login.Disabled=true
+AzureWebJobs.pupil-prefs.Disabled=true
+AzureWebJobs.report-psychometrician.Disabled=true
+AzureWebJobs.report-psychometrician-create-download.Disabled=true
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply functions/disable-functions.env]: It's useful for developers who don't want to run all the functions.

**Comment by GuyHarwood** [comment]
@sinhaamit1980 @activemq seeing the 2 same travis errors on this and #1084 
https://travis-ci.org/DFEAGILEDEVOPS/MTC/builds/515624163?utm_source=github_status&utm_medium=notification

**Comment by GuyHarwood** [comment]
Closing until we are happy with local testing behaviour

**Pull request merged**